src/audio/host-zephyr.c

```diff
@@ -999,20 +906,16 @@ static int host_params(struct comp_dev *dev,
 	return err;
 }
 
-static int host_prepare(struct comp_dev *dev)
+/* configure the DMA params and descriptors for host buffer IO */
+static int host_params(struct comp_dev *dev,
+		       struct sof_ipc_stream_params *params)
 {
-	struct host_data *hd = comp_get_drvdata(dev);
-	struct comp_buffer __sparse_cache *buf_c;
-	int ret;
-
-	comp_dbg(dev, "host_prepare()");
-
-	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
```

Comment:
you need to keep this here as well

Reply:
Our mission is finally remove host_prepare calling, right? if put comp_set_state() in host_prepare, how to make comp_set_state work?

Reply:
@btian1 we will never remove host_prepare(). it is absolutely needed for the host device with IPC3.

Reply:
got it, what we need to do is: avoid use device ops to perform host functionality, and at the same time, we need keep host as both ipc3 and ipc4 compatible.